### PR TITLE
refactored admission to avoid internal client references

### PR DESCRIPTION
**Pull request by deads2k**
Refactored admission to avoid internal client references.  This required switching to plugin initializers for them.  And that required some rewiring of the plugin initializers. 

Technically I can decouple from the other two commits, but I'm optimistic that those will go through easy. This is slightly move invasive, but I'd like to shoot for pre-christmas to avoid new admission plugins coming through and breaking bits.

@sttts @derekwaynecarr

### CODE REVIEW

File: plugin/pkg/admission/serviceaccount/admission.go [hunk 1]
```diff
@@ -128,21 +142,27 @@ func NewServiceAccount(cl clientset.Interface) *serviceAccount {
...
 
-	return &serviceAccount{
-		Handler: admission.NewHandler(admission.Create),
-		// TODO: enable this once we've swept secret usage to account for adding secret references to service accounts
-		LimitSecretReferences: false,
-		// Auto mount service account API token secrets
-		MountServiceAccountToken: true,
-		// Reject pod creation until a service account token is available
-		RequireAPIToken: true,
+	a.Run()
```
Comment (sttts) [inline plugin/pkg/admission/serviceaccount/admission.go]: a bit unexpected to find the `Run` in a setter. At least I would call `SetInternalClientSet` something like `InitWithClientSet` to make clear it is a one-time initialization.

File: pkg/kubeapiserver/admission/initializer.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		informers:      sharedInformers,
+		authorizer:     authz,
+	}
+}
+
+// Initialize checks the initialization interfaces implemented by each plugin
+// and provide the appropriate initialization data
+func (i pluginInitializer) Initialize(plugin admission.Interface) {
+	if wants, ok := plugin.(WantsInternalClientSet); ok {
+		wants.SetInternalClientSet(i.internalClient)
```
Comment (sttts) [inline pkg/kubeapiserver/admission/initializer.go]: This has side-effects (`Run`). I would prefer to split that part out as an explicit call to a `Run` method.

**Comment by sttts** [comment]
Overall the change looks reasonable. I don't like though that initialization (even with multiple initializers) is mixed with running a plugin.

File: plugin/pkg/admission/serviceaccount/admission.go (hunk 1 above)
Comment (deads2k) [inline plugin/pkg/admission/serviceaccount/admission.go]: > a bit unexpected to find the Run in a setter. At least I would call SetInternalClientSet something like InitWithClientSet to make clear it is a one-time initialization.

I agree that its weird, but its pre-existing.  I think we want to find a way to plumb a stop channel all the way down and a separate run step, but not part of this.

File: pkg/kubeapiserver/admission/initializer.go (hunk 2 above)
Comment (sttts) [inline pkg/kubeapiserver/admission/initializer.go]: Please add a TODO here if this is fixed in a follow-up.

**Comment by sttts** [comment]
Beside the TODO, lgtm

**Pull request merged**